Lib/test/test_codecs.py

```diff
@@ -716,6 +716,14 @@ def test_bug691291(self):
                          encoding=self.encoding) as reader:
             self.assertEqual(reader.read(), s1)
 
+    def test_removed_u_mode(self):
+        # "U" mode has been removed in Python 3.9
+        for mode in ("U", "rU", "r+U"):
+            with self.assertRaises(ValueError) as cm:
+                codecs.open(support.TESTFN, mode)
```

Comment:
Pass an `encoding`.

Reply:
Ooooh, the codecs module remains a mystery to me :-) Ok, I added an encoding and I added tests for "rt" mode.